src/parser/parser.yy

```diff
@@ -884,6 +885,7 @@ io_head_decl
   : INPUT_DECL      { $$ = AstIoType::input;      }
   | OUTPUT_DECL     { $$ = AstIoType::output;     }
   | PRINTSIZE_DECL  { $$ = AstIoType::printsize;  }
+  | LIMITSIZE_DECL  { $$ = AstIoType::limitsize;  }
```

Comment:
This does not seem to be at all related to I/O. Could it instead be a new AST type?

Reply:
Longterm we should rename io-directives to relation-directives. 
The reason is that we should not build two different directive infrastructures, i.e., one for IO and the other one for computation like limitsize.